Windows build

sujithvm opened the pull request:
Issue fix #17 , #18 . Thank you

sujithvm commented:
@stoewer  Please review this pull request when you are free. Thank you

sujithvm commented:
@stoewer Please activate Appveyor build for the project. Thank you

stoewer commented:
@sujithvm I enabled builds on appveyor.

sujithvm commented:
@stoewer Can you please review this pull request? Thank you

stoewer commented on src/main/java/org/gnode/nix/Block.java:
```diff
@@ -20,20 +24,20 @@
     }
 
-    private native void allocate();
+    private synchronized native void allocate();
```
Synchronize modifier can be removed, its presence suggests that the methods are thread-safe which is not the case.

stoewer commented on pom.xml:
```diff
@@ -161,6 +161,17 @@
+                    <forkCount>13</forkCount>
+                    <reuseForks>false</reuseForks>
+                    <testFailureIgnore>true</testFailureIgnore>
+                </configuration>
```
In order to prevent parallel execution I would try the following settings:

```xml
<parallel>suites</parallel>
<forkCount>1</forkCount>
<threadCount>1</threadCount>
```

stoewer commented:
Have you tried the `@NotThreadSafe` annotation for unit test cases?

stoewer commented on pom.xml:
```diff
@@ -161,6 +161,17 @@
+                <configuration>
+                    <forkCount>13</forkCount>
+                    <reuseForks>false</reuseForks>
+                    <testFailureIgnore>true</testFailureIgnore>
```
This disables test failures on all systems (even locally) can you set this via an environment variable which is only set true on appveyor?

sujithvm commented:
@stoewer I have made the following changes
1) removed synchronized from all native methods
2) removed `testFailureIgnore` from pom and set it to true only for Appveyor builds 
3) updated the Surefire configuration as suggested and used NotThreadSafe annotation but still the problem seems to persist ( https://ci.appveyor.com/project/stoewer/nix-java/build/0.1.6/job/e7wn6dgp2xx01xds#L1226 )
